common/utils.js

```diff
@@ -1339,6 +1339,33 @@
             }
         }
 
+        function overrideDownloadClickBehavior(containerElem) {
+            var downloadElements = containerElem[0].querySelectorAll(".deriva-url-validate");
+
+            downloadElements.forEach(function (el) {
+                angular.element(el).on('click', function (e) {
+                    e.preventDefault();
+
+                    var spinnerHTML = ' <span class="glyphicon glyphicon-refresh glyphicon-refresh-animate"></span>';
+                    //show spinner
+                    e.target.innerHTML += spinnerHTML;
+
+                    var dcctx = ConfigUtils.getContextJSON();
+                    // make a HEAD request to check if the user can fetch the file
+                    dcctx.server.http.head(e.target.href).then(function (response) {
+                        // fetch the file for the user
+                        $window.open(e.target.href, '_blank');
+                    }).catch(function (exception) {
+                        // If an error occurs while a user is trying to download the file, allow them to dismiss the dialog
+                        ErrorService.handleException(ERMrest.responseToError(exception), true);
+                    }).finally(function () {
+                        // remove the spinner
+                        e.target.innerHTML = e.target.innerHTML.slice(0, e.target.innerHTML.indexOf(spinnerHTML));
```

Comment:
It seems like in case of 401 this "finally" won't even run. We might want to remove this spinner in the catch block above. Was there a reason for doing this here and not in the catch-block before calling the `handleException`?

Reply:
This finally is triggering when I test this with an asset/file that returns a 404. Do you know of any assets that have a permission on them requiring a user to be logged in?

Reply:
https://dev.rebuildingakidney.org/chaise/record/#2/RNASeq:File/RID=16-1YDJ

Reply:
But even if this didn't have any issues, why finally? Why shouldn't we just hide the spinner in the catch block?

Reply:
If the request is successful (the user is authorized) we still need to hide the spinner, I didn't want to duplicate logic.